lib/internal/Magento/Framework/Module/DependencyChecker.php

```diff
@@ -89,17 +77,19 @@ public function checkDependenciesWhenEnableModules($toBeEnabledModules, $current
      * @param string[] $moduleNames list of modules to be enabled/disabled
      * @param string[] $enabledModules list of enabled modules assuming enable/disable succeeds
      * @return array
+     * @throws \Magento\Framework\Exception\LocalizedException
      */
     private function checkDependencyGraph($isEnable, $moduleNames, $enabledModules)
     {
-        $this->graph = $this->createGraph();
+        $fullModuleList = $this->loader->load();
+        $graph = $this->createGraph($fullModuleList);
         $dependenciesMissingAll = [];
         $graphMode = $isEnable ? Graph::DIRECTIONAL : Graph::INVERSE;
         foreach ($moduleNames as $moduleName) {
             $dependenciesMissing = [];
-            $paths = $this->graph->findPathsToReachableNodes($moduleName, $graphMode);
+            $paths = $graph->findPathsToReachableNodes($moduleName, $graphMode);
             $modules = array_merge(
-                array_keys($this->fullModuleList),
+                array_keys($fullModuleList),
```

Comment:
Looks like extracting array let’s calculation outside of the loop should speed up little bit

Reply:
Also little bit lower in array in the loop for checking enabled modules, believe it easily could be replaced to map and isset

Reply:
It changes semantics of code. i don't like such changes without real needs